booking-app/components/src/client/routes/booking/formPages/UserRolePage.tsx

```diff
@@ -136,21 +203,36 @@ export default function UserRolePage({
   useEffect(() => {
     if (
       !prevWatchedFieldsRef.current ||
-      prevWatchedFieldsRef.current.otherDepartment !==
-        watchedFields.otherDepartment
+      prevWatchedFieldsRef.current.otherDepartment !== watchedFields.otherDepartment ||
+      prevWatchedFieldsRef.current.school !== watchedFields.school ||
+      prevWatchedFieldsRef.current.otherSchool !== watchedFields.otherSchool
     ) {
       setFormData({ ...watchedFields });
       prevWatchedFieldsRef.current = watchedFields;
     }
   }, [watchedFields, setFormData]);
 
   const getDisabled = () => {
+    if (isOtherSchool(watchedFields.school)) {
+      return !watchedFields.otherSchool?.trim() || !watchedFields.otherDepartment?.trim() || !role;
+    }
+    
     if (showOther && !watchedFields.otherDepartment) {
       return true;
     }
+    
     return !role || !department;
   };
 
+  const hasSelectedSchool = !!(watchedFields.school && watchedFields.school !== "");
+  const isSchoolOther = isOtherSchool(watchedFields.school);
+  const isDepartmentSelected = !!department;
+  const isDepartmentStepComplete = isSchoolOther
```

Comment:
This ternary chain is a bit hard to read and reason about at a glance.
Maybe we can refactor this into clearer named variables or early-return logic so the intent is more obvious.

For example:
```
let isDepartmentStepComplete = false;

if (isSchoolOther) {
  isDepartmentStepComplete = !!(
    watchedFields.otherSchool?.trim() &&
    watchedFields.otherDepartment?.trim()
  );
} else if (showOther) {
  isDepartmentStepComplete = !!watchedFields.otherDepartment?.trim();
} else {
  isDepartmentStepComplete = isDepartmentSelected;
}
```